src/libraries/System.Private.CoreLib/src/System/IO/Path.Unix.cs

```diff
@@ -9,9 +9,36 @@ namespace System.IO
 {
     public static partial class Path
     {
-        public static char[] GetInvalidFileNameChars() => new char[] { '\0', '/' };
+        public static char[] GetInvalidFileNameChars() => new char[] {'\0', '/'};
 
-        public static char[] GetInvalidPathChars() => new char[] { '\0' };
+        public static char[] GetInvalidPathChars() => new char[] {'\0'};
+
+        // Checks if the given path is available for use.
+        public static bool Exists([NotNullWhen(true)] string? path)
+        {
+            try
+            {
+                if (string.IsNullOrEmpty(path))
+                {
+                    return false;
+                }
+
+                ReadOnlySpan<char> fullPath = GetFullPath(path).AsSpan();
+                return Interop.Sys.LStat(fullPath, out Interop.Sys.FileStatus _) == 0;
```

Comment:
thank you for using `lstat` here and keeping the consistency with `File.Exists` and `Directory.Exists` which return true if the link exists, but the file does not. It's another thing worth mentioning in the docs.